File: backend/routers/archive.py
from fastapi import APIRouter, HTTPException, status
from ..models import ArchiveBatch
from ..storage import comment_store, archive_store

router = APIRouter()

@router.post("/archive/{item_id:path}", response_model=ArchiveBatch, status_code=status.HTTP_200_OK)
def archive_resolved(item_id: str):
    resolved = comment_store.remove_resolved(item_id)
    if not resolved:
        raise HTTPException(status_code=400, detail="No resolved comments to archive")
    return archive_store.add_batch(item_id, resolved)

@router.get("/archive/{item_id:path}", response_model=list[ArchiveBatch])
def list_archive(item_id: str):
    return archive_store.list(item_id)

File: server/crates/revi-server/src/handlers/archive.rs
use axum::{
    extract::{Path, State},
    Json,
};

use crate::{
    error::AppError,
    models::ArchiveBatch,
    router::AppState,
};

pub async fn archive_resolved(
    State(s): State<AppState>,
    Path(item_id): Path<String>,
) -> Result<Json<ArchiveBatch>, AppError> {
    let resolved = s
        .comments
        .remove_resolved(&item_id)
        .map_err(AppError::Internal)?;

    if resolved.is_empty() {
        return Err(AppError::BadRequest(
            "no resolved comments to archive".to_owned(),
        ));
    }

    let batch = s
        .archive
        .add_batch(&item_id, resolved)
        .map_err(AppError::Internal)?;

    Ok(Json(batch))
}

pub async fn list_archive(
    State(s): State<AppState>,
    Path(item_id): Path<String>,
) -> Result<Json<Vec<ArchiveBatch>>, AppError> {
    let batches = s.archive.list(&item_id).map_err(AppError::Internal)?;
    Ok(Json(batches))
}
